thorlcr/graph/thgraphslave.cpp

```diff
@@ -571,30 +571,20 @@ void CSlaveActivity::serializeStats(MemoryBuffer &mb)
     // and updates the activity's 'stats'.
     // The callback fetches the current state of the activity's stats, called within a critical section ('statsCs'),
     // which the activity should use to protect any objects it uses whilst stats are being collected.
+
+    CRuntimeStatisticCollection serializedStats(stats.queryMapping());
```

Comment:
I think better and simpler would be

CRuntimeStatisticCollection serializedStats(inactiveStats);

remove the someInactiveStats variable.

The copy constructor for CRuntimeStatisticCollection should be fixed to copy nested stats (see set())

Reply:
The copy constructor effectively does a memcpy.  It doesn't map via the mapping:

```
        for (unsigned i=0; i <= num; i++)
            values[i].set(_other.values[i].get());
```

Reply:
@ghalliday - I've pushed a commit that unconditionally copies via ctor.. but let me know what u think re. efficiency loss.

Reply:
> The copy constructor effectively does a memcpy. It doesn't map via the mapping:
> 
> ```
>         for (unsigned i=0; i <= num; i++)
>             values[i].set(_other.values[i].get());
> ```

ok yes, and I guess a extra condition on nesting is nothing